inc/saiobject.h

```diff
@@ -239,6 +239,50 @@ sai_status_t sai_bulk_get_attribute(
         _Inout_ sai_attribute_t **attr_list,
         _Inout_ sai_status_t *object_statuses);
 
+/**
+ * @brief Queries attributes of objects in bulk based on a query condition.
+ *
+ * This function performs a bulk query on SAI objects of a specified type,
+ * based on the provided query condition. It retrieves the keys of the
+ * objects that match the query condition, their attributes, and statuses.
+ *
+ * @param[in] switch_id Switch object ID to perform the query.
+ * @param[in] object_type Type of SAI objects to query.
+ * @param[in] query_attr_count Count for query attributes.
+ * @param[in] query_attr_list List of query attributes.
+ * @param[inout] object_count Number of objects retrieved.
+ * @param[inout] object_key Array of object keys retrieved.
+ * @param[inout] attr_count List of attr_count. Caller passes the number
+ *    of attribute allocated in. Callee returns with the actual
+ *    number of attributes filled in. If the count is less than
+ *    needed, callee fills with the needed count and do not fill
+ *    the attributes. Callee also set the corresponding status to
+ *    #SAI_STATUS_BUFFER_OVERFLOW.
+ * @param[inout] attr_list List of attributes for every object. Caller is
+ *    responsible for allocating and freeing buffer for the attributes.
+ *    For list based attribute, e.g., s32list, objlist, callee should
+ *    assume the caller has not allocated the memory for the list and
+ *    should only to fill the count but not list. Then, caller
+ *    can use corresponding get_attribute to get the list.
+ *
+ * @param[inout] object_statuses Status for each object. If the object does
+ *    not exist, callee sets the corresponding status to #SAI_STATUS_INVALID_OBJECT_ID.
+ *    If the allocated attribute count is not large enough,
+ *    set the status to #SAI_STATUS_BUFFER_OVERFLOW.
+ *
+ * @return #SAI_STATUS_SUCCESS on success, #SAI_STATUS_BUFFER_OVERFLOW if lists size insufficient, failure status code on error
+ */
+sai_status_t sai_bulk_query_attribute(
+        _In_ sai_object_id_t switch_id,
+        _In_ sai_object_type_t object_type,
+        _In_ uint32_t query_attr_count,
+        _In_ const sai_attribute_t *query_attr_list,
```

Comment:
query_ prefix is not needed, just attr_count and attr_list

Reply:
> query_ prefix is not needed, just attr_count and attr_list

Thanks for your comments.
I relied here https://github.com/opencomputeproject/SAI/pull/1948#issuecomment-1907369064.